File: 869. Reordered_Power_of_2/solution.java
class Solution {
    public boolean reorderedPowerOf2(int n) {
        
        String str_n = String.valueOf(n);
        char[] sorted_n = str_n.toCharArray();
        Arrays.sort(sorted_n);
        
        long multi = 1;
        for(int i = 1 ; i < 31 ; i++){
            String str_multi = String.valueOf(multi);
            char[] sorted_multi = str_multi.toCharArray();
            Arrays.sort(sorted_multi);
            
            if(Arrays.equals(sorted_n, sorted_multi))
                return true;
            
            multi *= 2;
            
        }
            
        return false;
        
    }
}

File: 869. Reordered_Power_of_2/solution.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        
        string str_n = to_string(n);
        sort(str_n.begin(), str_n.end());
        
        long long multi = 1;
        for(int i = 0 ; i < 31 ; i++){
            string str_multi = to_string(multi);
            sort(str_multi.begin(), str_multi.end());
            if(str_n == str_multi)
                return true;
            multi *= 2;
        }
        return false;
        
    }
};